src/common/types/metadata.js

```diff
@@ -114,7 +114,29 @@ type MetadataSuggestion = {
     suggestions: { [key: string]: string | number | string[] },
 };
 
+type MetadataInstanceTemplateField = {
+    description?: string,
+    displayName: string,
+    hidden?: boolean,
+    id: string,
+    key: string, // V2
+    options?: Array<MetadataTemplateFieldOption>, // V3
+    type: MetadataFieldType,
+    value: string,
+};
+
+type MetadataTemplateInstance = {
+    displayName?: string,
+    hidden?: boolean,
+    id: string,
+    metadataFields: MetadataInstanceTemplateField[],
+    scope: string,
+    templateKey: string,
+};
+
 export type {
+    MetadataInstanceTemplateField,
+    MetadataTemplateInstance,
```

Comment:
want to confirm if the order of `InstanceTemplate` and `TemplateInstance` are correct

Reply:
OH, I messed that up. Thanks for in depth review. I am fixing that :)